### PR TITLE
fix issue for bfv deploy on zvm not support multipath ipl

### PR DESCRIPTION
**Pull request by haolp**

### CODE REVIEW

**Comment by jichenjc** [comment]
@haolp can you help add UT for this ?

**Comment by haolp** [comment]
> @haolp can you help add UT for this ?

sure, thanks for your reminder.

File: smtLayer/makeVM.py [hunk 1]
```diff
@@ -152,7 +152,10 @@ def createVM(rh):
         dirLines.append("COMMAND ATTACH PCIF %s * AS %s" % (s[0], s[1]))
 
     if 'ipl' in rh.parms:
-        if 'loadportname' in rh.parms and 'loadlun' in rh.parms:
+        # only use IPL LOADDEV for BFV deploy when the z/VM supports multipath IPL
+        # 'alterdev' in rh.parms means the z/VM supports multipath IPL
+        if ('alterdev' in rh.parms and 'loadportname' in rh.parms and
```
Comment (dongyanyang) [inline smtLayer/makeVM.py:157]: change the condition to check z/VM version.

**Review by dongyanyang — CHANGES_REQUESTED** [review]

File: zvmsdk/utils.py [hunk 2]
```diff
@@ -1202,3 +1202,30 @@ def get_lpar_name(zhypinfo=None):
         zhypinfo = get_zhypinfo(filter='all')
         lpar_name = zhypinfo['lpar']['layer_name']
     return lpar_name
+
+
+def _getZVMVersion():
```
Comment (dongyanyang) [inline zvmsdk/utils.py:1207]: please follow the same naming conventions of existing functions.

def get_zvm_version()

File: zvmsdk/utils.py [hunk 3]
```diff
@@ -1202,3 +1202,30 @@ def get_lpar_name(zhypinfo=None):
...
+    cmdOutputLines = output.split("\n")
+    # firstLine = 'z/VM Version 7 Release 3.0, service level 0000 (64-bit)'
+    firstLine = cmdOutputLines[0]
+    result = firstLine.split()
+    bigV = float(result[2])
+    smallV = float(result[4].strip(','))
+    return (rc, bigV, smallV)
+
+
+def isSupportLoaddev():
```
Comment (dongyanyang) [inline zvmsdk/utils.py:1226]: is_ipl_loaddev_supported

**Review by dongyanyang — CHANGES_REQUESTED** [review]

File: zvmsdk/utils.py [hunk 4]
```diff
@@ -1202,3 +1202,30 @@ def get_lpar_name(zhypinfo=None):
...
+        LOG.warning(msg)
+    # output = 'z/VM Version 7 Release 3.0, service level 0000 (64-bit)\n
+    # Generated at 07/20/23 03:29:47 CEST\n
+    # IPL at 07/20/23 10:42:17 CEST\n'
+    cmdOutputLines = output.split("\n")
+    # firstLine = 'z/VM Version 7 Release 3.0, service level 0000 (64-bit)'
+    firstLine = cmdOutputLines[0]
+    result = firstLine.split()
+    bigV = float(result[2])
+    smallV = float(result[4].strip(','))
```
Comment (dongyanyang) [inline zvmsdk/utils.py:1222]: these match rules are too risky, I think.
I have other options:
1. smcli - then you can just get the last string of the output.
```
[root@quality-ssi001 ~]# smcli Query_API_Functional_Level -T SSI10289
The API functional level is z/VM V7.2
```
2. from zthin file:
```
[root@quality-ssi001 ~]# cat /var/opt/zthin/smapi.level
720
```

Or @jichenjc @bjhuangr Do you have suggestions on how to safely get and compare the z/VM version?

File: zvmsdk/utils.py (hunk 3 above)
Comment (haolp) [reply zvmsdk/utils.py]: updated, thanks.

File: smtLayer/makeVM.py (hunk 1 above)
Comment (haolp) [reply smtLayer/makeVM.py]: updated, thanks.

File: zvmsdk/utils.py (hunk 4 above)
Comment (bjhuangr) [reply zvmsdk/utils.py:1222]: it's fine to use `/var/opt/zthin/smapi.level`

Comment (haolp) [reply zvmsdk/utils.py]: thanks,  design changed and do not need to check zvm version now.

**Review by dongyanyang — APPROVED** [review]

**Review by bjhuangr — APPROVED** [review]

**Pull request merged**